ssl/statem/statem_lib.c

```diff
@@ -2598,6 +2598,32 @@ int check_in_list(SSL_CONNECTION *s, uint16_t group_id, const uint16_t *groups,
     return 0;
 }
 
+/*
+ * Same as above, but also provides position where group was found
+ * Returns 1 if group is found, returns 0 if group was not found
+ */
+int check_in_list_pos(SSL_CONNECTION *s, uint16_t group_id, const uint16_t *groups,
+                      size_t num_groups, int checkallow, size_t *pos)
+{
+    size_t i;
+
+    if (groups == NULL || num_groups == 0)
+        return 0;
+
+    for (i = 0; i < num_groups; i++) {
+        uint16_t group = groups[i];
+
+        if (group_id == group
+                && (!checkallow
+                    || tls_group_allowed(s, group, SSL_SECOP_CURVE_CHECK))) {
+            *pos = i;
```

Comment:
Perhaps support `pos` being NULL, and then don't return it, and then the above function can just become a wrapper than passes NULL, or just passes the address of a throw-away variable.